src/include/txn.i

```diff
@@ -511,9 +511,14 @@ __wt_txn_modify(WT_SESSION_IMPL *session, WT_UPDATE *upd)
 
 	txn = &session->txn;
 
-	if (F_ISSET(txn, WT_TXN_READONLY))
+	if (F_ISSET(txn, WT_TXN_READONLY)) {
+		if (F_ISSET(txn, WT_TXN_IGNORE_PREPARE))
```

Comment:
@luke-pearson It looks like this returns  ENOTSUP even when `ignore_prepare=force`. Is there more work needed to suppress the error in that case? I applied this locally with the server changes, but I still get failures.

Reply:
I've put in a fix for the issue, could you please retest?

Reply:
This does work now, thanks.